File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs == null || strs.length == 0)
        {
            return new ArrayList<>();
        }
        HashMap<String, List<String>> map = new HashMap<>();
        for (String s : strs)
        {
            char a[] = new char[26];
            for (char c : s.toCharArray())
            {
                a[c - 'a']++;
            }
            String str = String.valueOf(a);
            if (!map.containsKey(str))
            {
                map.put(str, new ArrayList<>());
            }
            map.get(str).add(s);
        }
        return new ArrayList<>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        hashmap = {}
        output = []
        for st in strs:
            sorted_list = sorted(st)
            sorted_str = "".join(sorted_list)
            if sorted_str in hashmap:
                hashmap[sorted_str].append(st)
            else:
                hashmap[sorted_str] = [st]
        for key in hashmap:
            output.append(hashmap[key])
        return output